test/ecto/enum_test.exs

```diff
@@ -78,7 +79,7 @@ defmodule Ecto.EnumTest do
                   %{
                     on_dump: %{bar: "baar", baz: "baaz", foo: "fooo"},
                     on_load: %{"baar" => :bar, "baaz" => :baz, "fooo" => :foo},
-                    values: [:foo, :bar, :baz],
+                    mappings: [foo: "fooo", bar: "baar", baz: "baaz"],
```

Comment:
Would map be a more reasonable data structure here? Because keyword would let you do something like this: `[foo: "foo", foo: "fooo"]` and the first one would win.

Reply:
I think we can detect and raise on duplicate keys. I can see how using list and thus preserving order can be useful for reflection purposes but no strong opinion either way.

Reply:
My reason to chose a keyword list over a map is the fact that the user inputs a keyword list when creating a Ecto.Enum field, so it seemed consistent returning the exact same thing the user inputted. Yet, as you pointed out, this didn't reflect the exact mappings that are done on `on_dump` and `on_load` (these are maps, and the last one would take precedence). I opened another PR that prevents the user from inputting keyword lists with duplicate keys.